doc/kernel_chaos.md

```diff
@@ -0,0 +1,86 @@
+# Kernel Chaos Document
+
+This document describe how to add kernel chaos experiments in Chaos Mesh.
+
+Kernel chaos's function is similar to
+[inject.py](https://github.com/iovisor/bcc/blob/master/tools/inject.py), which
+guarantees the appropriate erroneous return of the specified injection mode
+(kmalloc,bio,etc) given a call chain and an optional set of predicates.
+
+You can read
+[inject\_example.txt](https://github.com/iovisor/bcc/blob/master/tools/inject_example.txt)
+to learn more.
+
+Below is a sample kernel chaos configuration file:
+
+```yaml
+apiVersion: pingcap.com/v1alpha1
+kind: KernelChaos
+metadata:
+  name: kernel-chaos-example
+  namespace: chaos-testing
+spec:
+  mode: one
+  selector:
+    namespaces:
+      - chaos-mount 
+  failKernRequest:
+    callchain:
+        - funcname: "__x64_sys_mount"
+    failtype: 0
+```
+
+And a sample program:
+```c
+#include <sys/mount.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <unistd.h>
+
+int main(void) {
+	int ret; 
+	while (1) {
+		ret = mount("/dev/sdc", "/mnt", "ext4", 
+			    MS_MGC_VAL | MS_RDONLY | MS_NOSUID, "");
+		if (ret < 0)
+			fprintf(stderr, "%s\n", strerror(errno));
+		sleep(1);
+		ret = umount("/mnt");
+		if (ret < 0)
+			fprintf(stderr, "%s\n", strerror(errno));
+	}
+}
+```
+
+During the injection, the program will output:
+
+> Cannot allocate memory
+> Invalid argument
+> Cannot allocate memory
+> Invalid argument
+> Cannot allocate memory
+> Invalid argument
+> Cannot allocate memory
+> Invalid argument
+> Cannot allocate memory
+> Invalid argument
+
+> For more sample files, see [examples](https://github.com/chaos-mesh/bpfki/tree/develop/examples). You can edit them as needed. 
+
+Description:
+
+* **mode** defines the mode to select pods.
+* **selector** specifies the target pods for chaos injection.
+* **failkernRequest** defines the specified injection mode (kmalloc,bio,etc) with a call chain and an optional set of predicates.
```

Comment:
What is the difference between `kmalloc` and `bio`?

Reply:
`bio` is a concept of block layer, so it's used to inject io request error. 
`kmalloc` is the mostly caller of `slab`.